src/drivers/QtTestDriver.cpp

```diff
@@ -1,201 +1,30 @@
-            fd_blocked = false;
-#ifdef WINDOWS
-            if (!eof(m_pipe[READ]))
-                bytesRead = read(m_pipe[READ], buf, bufSize-1);
-#else
-            bytesRead = read(m_pipe[READ], buf, bufSize-1);
-#endif
-            if (bytesRead > 0) {
-                buf[bytesRead] = 0;
-                m_captured += buf;
-            } else if (bytesRead < 0) {
-                fd_blocked = (errno == EAGAIN || errno == EWOULDBLOCK || errno == EINTR);
-                if (fd_blocked)
-                    QThread::msleep(10);
-            }
-        }
-        while(fd_blocked || bytesRead == (bufSize-1));
-
-        secure_close(m_oldStdOut);
-        secure_close(m_oldStdErr);
-        secure_close(m_pipe[READ]);
-#ifdef WINDOWS
-        secure_close(m_pipe[WRITE]);
-#endif
-        m_capturing = false;
-    }
-
-    std::string GetCapture() {
-        QMutexLocker locker(&m_mutex);
-        return m_captured;
-    }
-
-private:
-    enum PIPES { READ, WRITE };
-
-    int secure_dup(int src) {
-        int ret = -1;
-        bool fd_blocked = false;
-        do {
-             ret = dup(src);
-             fd_blocked = (errno == EINTR ||  errno == EBUSY);
-             if(fd_blocked)
-                 QThread::msleep(10);
-        } while (ret < 0);
-        return ret;
-    }
-
-    void secure_pipe(int * pipes) {
-        int ret = -1;
-        bool fd_blocked = false;
-        do {
-#ifdef WINDOWS
-            ret = pipe(pipes, 65536, O_BINARY);
-#else
-            ret = pipe(pipes) == -1;
-#endif
-            fd_blocked = (errno == EINTR ||  errno == EBUSY);
-            if (fd_blocked)
-                QThread::msleep(10);
-        } while (ret < 0);
-    }
-
-    void secure_dup2(int src, int dest) {
-        int ret = -1;
-        bool fd_blocked = false;
-        do {
-             ret = dup2(src,dest);
-             fd_blocked = (errno == EINTR ||  errno == EBUSY);
-             if (fd_blocked)
-                QThread::msleep(10);
-        } while (ret < 0);
-    }
-
-    void secure_close(int & fd) {
-        int ret = -1;
-        bool fd_blocked = false;
-        do {
-             ret = close(fd);
-             fd_blocked = (errno == EINTR);
-             if (fd_blocked)
-                QThread::msleep(10);
-        } while (ret < 0);
-
-        fd = -1;
-    }
-
-    int m_pipe[2];
-    int m_oldStdOut;
-    int m_oldStdErr;
-    bool m_capturing;
-    QMutex m_mutex;
-    std::string m_captured;
-};
+#include <QTextStream>
+#include <QTemporaryFile>
 
 namespace cucumber {
 namespace internal {
 
 const InvokeResult QtTestStep::invokeStepBody() {
-    QtTestObject testObject(this);
-
-    qtCapture capture;
-    capture.BeginCapture();
-    int returnValue = QTest::qExec(&testObject, 0, NULL);
-    capture.EndCapture();
+    QTemporaryFile file;
+    QString fileName;
+    if(!file.open()) {
+        return InvokeResult::failure("Unable to open temporary file needed for this test");
+    }
+    file.close();
```

Comment:
It doesn't destroy used name, but it turns out to be unnecessary to close and reopen.